File: Grind169/Week2/409-Longest-Palindrome/solution.cpp
#include <unordered_map>
#include <string>

using namespace std;

class Solution
{
public:
    int longestPalindrome(string s)
    {
        unordered_map<char, bool> map;
        int res = 0;

        for (int i = 0; i < s.size(); i++)
        {
            if (map.find(s[i]) == map.end() || map[s[i]] == false)
            {
                map[s[i]] = true;
            }
            else
            {
                map[s[i]] = false;
                res += 2;
            }
        }

        for (auto &n : map)
        {
            if (n.second == true)
            {
                return res + 1;
            }
        }

        return res;
    }
};

File: Grind169/Week2/409-Longest-Palindrome/solution.py
class Solution:
    def longestPalindrome(self, s: str) -> int:
        res = 0
        map = {}

        for ch in s:
            if map.get(ch) is None:
                map[ch] = 1
            else:
                map.pop(ch)
                res += 2

        if map:
            res += 1

        return res
